ts/parts/Pointer.ts

```diff
@@ -1002,7 +1002,10 @@ Highcharts.Pointer.prototype = {
                 } else if (hoverPoint) { // #2500
                     hoverPoint.setState(hoverPoint.state, true);
                     chart.axes.forEach(function (axis: Highcharts.Axis): void {
-                        if (axis.crosshair) {
+                        if (
+                            axis.crosshair &&
+                            (hoverPoint as any).series[axis.coll] === axis
```

Comment:
Added casting to any similar to this one:
https://github.com/highcharts/highcharts/blob/0db656018c3a976d7e8ac38e9ff4d4778256603a/ts/parts/Pointer.ts#L868-L870

The difference here is `(hoverPoint as any).series` vs `(hoverPoint.series as any)`. For some reason TS said that `hoverPoint` can be undefined, which clearly is checked above. Did I miss something?

Reply:
That is typical for TypeScript, if you use an outer variable in an inner function. TypeScript is not able to determ, when the inner function gets called and the outer variable might change. Therefor it has to assume, that every of the possible variable types can be possible in the inner function. So you have to do an other validate check.

That said, in this case a cast to any is still necessary, because you use hoverPoint with an indexer, which the Point class does not have. So a inner validation does not help in type sense.

Reply:
In few words: The any cast is fine as it is for now.

Reply:
Thanks for the explanation!